lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -453,6 +453,7 @@ sub job_next_previous_ajax ($self) {
                 DT_RowId => 'job_result_' . $job_id,
                 id => $job_id,
                 name => $job->name,
+                assigned_worker => $job->assigned_worker->name,
```

Comment:
Have you tested the performance impact? This will do yet another SQL query for every job for this already slow route. The "prefetching" I meant in another comment is basically a join with the workers table when querying jobs. I suppose the "prefetching" needed to be done in https://github.com/os-autoinst/openQA/blob/9cf9ed8fc9feb80490887185df8d0af2c0c5b603/lib/OpenQA/Schema/ResultSet/Jobs.pm#L399 using some DBIx magic. Maybe this doesn't work in this special case, though. Then you'd likely need to touch the raw SQL code in https://github.com/os-autoinst/openQA/blob/9cf9ed8fc9feb80490887185df8d0af2c0c5b603/lib/OpenQA/Schema/Result/JobNextPrevious.pm.

Reply:
Not sure how I could measure the perf impact.

The SQL code has already `SELECT *`, so I think it should be pre-fetched already, no?

Reply:
No, it needed a join or a nested query with the workers table to return the associated worker name. However, I'd needed to play around with this myself because using a view is a special case.

For performance testing itself, I usually import a production database locally. This isn't an option for your but you could simply create lots of jobs (that are next/previous ones for an existing job) manually. To measure the response time of web UI routes I usually use a Mojolicious plugin (see https://github.com/Martchus/openQA-helper#profiling).

Note that I'm only concerned about this because this route already takes long anyways (e.g. checkout https://openqa.opensuse.org/tests/3126056#next_previous) and if we'd multiply this time even by a slight factor due to an additional/separate query per job this feature might do more harm than good.